### ISSUE TITLE
remove `logUncaughtExceptions` config var and print uncaught exceptions to stderr by default

### ISSUE THREAD

**Issue by trentm** [elastic/apm-agent-nodejs#2412]
By default the APM agent installs an `uncaughtException` handler to send an error to APM server. By default, so as to not change application behaviour, it exits the process after sending the error.

A side-effect of installing an `uncaughtException` handler is that node core no longer does its full default behaviour if there are no such handlers. Its default behaviour is to (a) print the error to stderr and (b) exit (modulo `--abort-on-uncaught-exception`).

If the `logUncaughtExceptions: true` config option is given to the APM agent, then it will also print the error to stderr. 
https://www.elastic.co/guide/en/apm/agent/nodejs/current/configuration.html#log-uncaught-exceptions
`logUncaughtExceptions` defaults to false. I think it should default to true.

Note https://github.com/elastic/apm-agent-nodejs/issues/2367 as well. If we switch to using uncaughtExceptionMonitor (in node v12.7.0 and greater) and that solves this issue, then great. Perhaps, then, this issue will only be relevant for earlier node versions.

**Update 2023-08-08:** Using `uncaughtExceptionMonitor` is promising but too much work for now. We can just **drop** the `logUncaughtExceptions` config option instead, and print a captured uncaught exception to stderr by default.

**Comment by trentm** [comment]
> Note #2367 as well. If we switch to using uncaughtExceptionMonitor (in node v12.7.0 and greater) and that solves this issue, then great. Perhaps, then, this issue will only be relevant for earlier node versions.

See https://github.com/elastic/apm-agent-nodejs/issues/2367#issuecomment-1670408202
tl;dr: I think switching to using `uncaughtExceptionMonitor` -- instead of `uncaughtException` -- would be a good thing eventually, but not now for the 4.0.0 milestone because of development effort involved.
